web/src/app/util/SpeedDial.js

```diff
@@ -1,74 +1,61 @@
-import React from 'react'
+import React, { useState } from 'react'
 import p from 'prop-types'
-import SpeedDialAction from '@material-ui/lab/SpeedDialAction'
-import SpeedDialIcon from '@material-ui/lab/SpeedDialIcon'
-import SpeedDial from '@material-ui/lab/SpeedDial'
+import { SpeedDial, SpeedDialIcon, SpeedDialAction } from '@material-ui/lab'
 
-export default class ScheduleNewOverrideFAB extends React.PureComponent {
-  static propsTypes = {
-    label: p.string.isRequired,
-    actions: p.arrayOf(
-      p.shape({
-        icon: p.element.isRequired,
-        onClick: p.func.isRequired,
-        label: p.string.isRequired,
-      }),
-    ).isRequired,
-  }
+export default function CustomSpeedDial(props) {
+  const [open, setOpen] = useState(false)
 
-  state = {
-    open: false,
-  }
+  const doToggle = () => setOpen(!open)
+  const doOpen = () => setOpen(true)
+  const doClose = () => setOpen(false)
 
-  shownState = false
-  _shownTimeout = -1
+  return (
+    <SpeedDial
+      ariaLabel={props.label}
+      FabProps={{
+        'data-cy': 'page-fab',
+      }}
+      icon={<SpeedDialIcon />}
+      onClick={doToggle}
+      onClose={doClose}
+      onMouseEnter={doOpen}
+      onMouseLeave={doClose}
+      open={open}
+      style={{
+        position: 'fixed',
+        bottom: '2em',
+        right: '2em',
+        zIndex: 9001,
+      }}
+    >
+      {props.actions
+        .slice() // TODO why this?
+        .reverse() // TODO why this? If we cut this line, we have to reorder the action arrays everywhere else (3 instances)
```

Comment:
Not sure why we are reversing, but I imagine the slice is so we aren't modifying the original array when we reverse. Can probably delete both, but we'll have to update the order where we're using this component